src/System.Collections.Concurrent/tests/BlockingCollectionTests.cs

```diff
@@ -48,5 +48,1967 @@ public static void TestBlockingCollectionBasic()
+            BlockingCollection<int> blockingCollection = ConstructBlockingCollection<int>();
+
+            blockingCollection.Add(0);
+
+            blockingCollection.CompleteAdding();
+
+            try
+            {
+                blockingCollection.Add(1);
+                Assert.False(true, "RunBlockingCollectionTest7_CompleteAdding: > test failed - Add should have thrown InvalidOperationException");
+            }
+            catch (InvalidOperationException)
+            {
+            }
+
+            if (blockingCollection.Count != 1)
+            {
+                Assert.False(true, String.Format("RunBlockingCollectionTest7_CompleteAdding: > test failed - Unexpected count: Actual={0}, Expected=1", blockingCollection.Count));
+            }
+
+            blockingCollection.Take();
+
+            try
+            {
+                blockingCollection.Take();
+                Assert.False(true, "RunBlockingCollectionTest7_CompleteAdding: > test failed - Take should have thrown OperationCanceledException");
+            }
+            catch (InvalidOperationException)
+            {
+            }
+
+            int item = 0;
+
+            if (blockingCollection.TryTake(out item))
+            {
+                Assert.False(true, "RunBlockingCollectionTest7_CompleteAdding: > test failed - TryTake should have return false");
+            }
+
+            int counter = 0;
+            foreach (int i in blockingCollection.GetConsumingEnumerable())
+            {
+                counter++;
+            }
+
+            if (counter > 0)
+            {
+                Assert.False(true, "RunBlockingCollectionTest7_CompleteAdding: > test failed - the enumerable returned from GetConsumingEnumerable() should not have enumerated through the collection");
+            }
+        }
+
+        [Fact]
+        [OuterLoop]
+        public static void RunBlockingCollectionTest7_ConcurrentAdd_CompleteAdding()
+        {
+            BlockingCollection<int> blockingCollection = ConstructBlockingCollection<int>();
+            Task[] threads = new Task[4];
+            int succeededAdd = 0;
+            for (int i = 0; i < threads.Length; i++)
+            {
+                threads[i] = Task.Run(() =>
+                {
+                    for (int j = 0; j < 1000; j++)
+                    {
+                        try
+                        {
+                            blockingCollection.Add(j);
+                            Interlocked.Increment(ref succeededAdd);
+                        }
+                        catch (InvalidOperationException)
+                        {
+                            break;
+                        }
+
+                    }
+                });
+            }
+
+            blockingCollection.CompleteAdding();
+            int count1 = blockingCollection.Count;
+            Task.WaitAll(Task.Delay(100));
+            int count2 = blockingCollection.Count;
+
+            if (count1 != count2)
```

Comment:
Instead of "if (A != B) Assert.False(true, ...)", couldn't it just be "Assert.NotEqual(A, B)" or something similar?